libraries/AP_GPS/AP_GPS_GSOF.cpp

```diff
@@ -260,7 +260,7 @@ AP_GPS_GSOF::process_message(void)
               pageidx, maxpageidx, trans_number);
 #endif
 
-        int valid = 0;
```

Comment:
Based on "The only memory you MUST zero is stack-stored variables - locals", this is probably bad.